Geometry/CSCGeometryBuilder/src/CSCGeometryParsFromDD.cc

```diff
@@ -426,24 +452,31 @@ bool CSCGeometryParsFromDD::build(const cms::DDCompactView* cview,
       }
 
       auto wirespacing = fv.get<double>("WireSpacing");
-      wg.wireSpacing = static_cast<double>(wirespacing);
+      wg.wireSpacing = static_cast<double>(wirespacing / dd4hep::cm);
```

Comment:
I suggest simplifying the treatment of `wireSpacing`. Convert it to mm at the start.
```
wg.wireSpacing = fv.get<double>("WireSpacing") / dd4hep::mm;
```
In the debug output, give "mm" as its units. And then in line 479, there is no conversion because it is already in mm.

Reply:
@cvuosalo : wg.wireSpacing is read, by DD4HEP, in "cm" (see PR description) so in line 479 is neessary to put / dd4hep::mm.

Reply:
This code goes through three steps with wire spacing:
```
auto wirespacing = fv.get<double>("WireSpacing");           // Get wire spacing in DD4hep units
wg.wireSpacing = static_cast<double>(wirespacing / dd4hep::cm);  // Convert wire spacing to cm
uparvals.emplace_back(wg.wireSpacing / dd4hep::mm);             // Store wire spacing in mm
```
I don't see any reason for the intermediate conversion to cm. Why is that needed? Don't convert it to cm. It could be converted to mm in the first step in line 456, and that would be the only conversion:
```
wg.wireSpacing = fv.get<double>("WireSpacing") / dd4hep::mm;
```

Reply:
@cvuosalo : before adding "/ dd4hep::cm" in the line "wg.wireSpacing = static_cast<double>(wirespacing / dd4hep::cm);" I have checked the printouts written inside the myLog.lo file and I saw that for the DD4HEP part wirespacing was written in "cm" not in "mm" (i.e without "/dd4hep::cm"). All parameters inside the DD4HEP part are in "cm". So I added "/dd4hep::cm" in order to put the unit in the code for a future fix. I noted that adding "/dd4hep::cm" in all the parameters that are already "cm" does not change the value.  The only part of the DD4HEP code written in "mm" was "uparvals.emplace_back((wg.wireSpacing) * 10.0);" in line 446. This is why I wrote /dd4hep::cm and /dd4hep::mm in the code.

Reply:
Yes, the code here is designed in an inconsistent way. `alignmentPinToFirstWire`, `narrowWidthOfWirePlane`, and others are stored in cm. However, `wireSpacing` is stored in mm. I am suggesting to make a change so `wireSpacing` is changed to mm at the first step, which will require changing the debug print-out statements so they say that `wireSpacing` is in mm.
The reason we are changing the DD4hep units is to reduce the number of multiplications. Since `wireSpacing` is stored in mm, it never needs to be in cm, so let's eliminate the step that converts `wireSpacing` to cm.